### PR TITLE
Use tril() and triu() function from dpctl.tensor

### PR DESCRIPTION
**Pull request by npolina4**
Use tril() and triu() functions from dpctl.tensor module instead of DPNP backend implementation.
A cython code relating to tril() and triu() call is cleaned up.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_arraycreation.py [hunk 1]
```diff
@@ -1350,12 +1356,14 @@ def tril(x1, k=0):
...
-    if x1_desc:
-        if not isinstance(k, int):
-            pass
-        else:
-            return dpnp_tril(x1_desc, k).get_pyobj()
+    if not dpnp.isarray(x1):
+        pass
+    elif x1.ndim < 2:
+        pass
+    elif not isinstance(k, int):
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_arraycreation.py:1363]: I think it'd be better to implement the same check through `operator.index` like dpctl did.
Then the following code will be supported:
```
a = dpnp.arange(9, dtype=dpnp.int64)
dpnp.tril(a.reshape((3, 3)), k=a[0])
```

File: tests/skipped_tests.tbl [hunk 2]
```diff
@@ -430,6 +430,9 @@ tests/third_party/cupy/creation_tests/test_ranges.py::TestRanges::test_linspace_
 tests/third_party/cupy/creation_tests/test_ranges.py::TestRanges::test_linspace_mixed_start_stop
 tests/third_party/cupy/creation_tests/test_ranges.py::TestRanges::test_linspace_mixed_start_stop2
 tests/third_party/cupy/creation_tests/test_ranges.py::TestRanges::test_linspace_start_stop_list
+tests/third_party/cupy/creation_tests/test_matrix.py::TestTriLowerAndUpper_param_0_{shape=(2,)}::test_tril
```
Comment (antonwolfy) [inline tests/skipped_tests.tbl:433]: why do you mark the tests as to be skipped?

File: tests/test_arraycreation.py [hunk 3]
```diff
@@ -258,28 +258,25 @@ def test_tri_default_dtype():
...
-                              '[[0, 0], [0, 0]]',
+                         ids=['[[0, 0], [0, 0]]',
                               '[[1, 2], [1, 2]]',
                               '[[1, 2], [3, 4]]',
                               '[[0, 1, 2], [3, 4, 5], [6, 7, 8]]',
                               '[[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]]'])
-def test_tril(m, k):
-    a = numpy.array(m)
+@pytest.mark.parametrize("dtype", get_all_dtypes(no_float16=False))
+def test_tril(m, k, dtype):
```
Comment (antonwolfy) [inline tests/test_arraycreation.py:275]: I think it'd be great also to add tests with `dpnp.tril` in `test_sycl_queue.py` and `test_usm_type.py` to secure CFD.

File: dpnp/dpnp_iface_arraycreation.py [hunk 4]
```diff
@@ -1331,14 +1331,20 @@ def tri(N, M=None, k=0, dtype=dpnp.float, **kwargs):
...
     Lower triangle of an array.
 
     Return a copy of an array with elements above the `k`-th diagonal zeroed.
 
     For full documentation refer to :obj:`numpy.tril`.
 
+    Limitations
+    -----------
+    Parameter ``x1`` is supported only as :class:`dpnp.dpnp_array` with two or more dimensions.
+    Parameter ``k`` is supported only as int data type.
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_arraycreation.py:1345]:
```suggestion
    Parameter `k` is supported only of integer data type.
```

File: dpnp/dpnp_iface_arraycreation.py [hunk 5]
```diff
@@ -1331,14 +1331,20 @@ def tri(N, M=None, k=0, dtype=dpnp.float, **kwargs):
...
     """
     Lower triangle of an array.
 
     Return a copy of an array with elements above the `k`-th diagonal zeroed.
 
     For full documentation refer to :obj:`numpy.tril`.
 
+    Limitations
+    -----------
+    Parameter ``x1`` is supported only as :class:`dpnp.dpnp_array` with two or more dimensions.
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_arraycreation.py:1344]:
```suggestion
    Parameter `x1` is supported only as :class:`dpnp.dpnp_array` or :class:`dpctl.tensor.usm_ndarray` with two or more dimensions.
```

File: dpnp/dpnp_container.py [hunk 6]
```diff
@@ -200,6 +201,12 @@ def ones(shape,
     return dpnp_array(array_obj.shape, buffer=array_obj, order=order)
 
 
+def tril(x1, /, *, k=0):
+    """"Creates `dpnp_array` as lower triangle of an input array."""
```
Comment (antonwolfy) [inline dpnp/dpnp_container.py:205]:
```suggestion
    """"Creates `dpnp_array` as lower triangular part of an input array."""
```

File: tests/skipped_tests.tbl (hunk 2 above)
Comment (npolina4) [reply tests/skipped_tests.tbl:433]: dpctl.tensor.tril() does not support one dimension array.
Array API standard definition:
x (array) – input array having shape (..., M, N) and whose innermost two dimensions form MxN matrices.

File: dpnp/dpnp_iface_arraycreation.py (hunk 1 above)
Comment (npolina4) [reply dpnp/dpnp_iface_arraycreation.py:1363]: I try and see:
```python
a = dpnp.arange(9, dtype=dpnp.int64)
operator.index(a[0])
TypeError: 'dpnp_array' object cannot be interpreted as an integer
```

Comment (antonwolfy) [reply dpnp/dpnp_iface_arraycreation.py:1363]: It will be supported once PR #1285 is merged.

File: tests/test_sycl_queue.py [hunk 7]
```diff
@@ -161,6 +161,16 @@ def test_array_creation_like(func, kwargs, device_x, device_y):
...
 
 
+@pytest.mark.parametrize("func", ["tril", "triu"], ids=["tril", "triu"])
+@pytest.mark.parametrize("device",
+                          valid_devices,
+                          ids=[device.filter_string for device in valid_devices])
+def test_tril_triu(func, device):
+    x0 = dpnp.ones((3,3), device=device)
+    x = getattr(dpnp, func)(x0)
+    assert x.sycl_device == device
```
Comment (antonwolfy) [inline tests/test_sycl_queue.py:171]:
```suggestion
    assert_sycl_queue_equal(x.sycl_queue, x0.sycl_queue)
```

File: dpnp/dpnp_container.py [hunk 8]
```diff
@@ -200,6 +203,22 @@ def ones(shape,
     return dpnp_array(array_obj.shape, buffer=array_obj, order=order)
 
 
+def tril(x1, /, *, k=0):
+    k = operator.index(k)
+    order = "F" if (x1.flags.f_contiguous) else "C"
```
Comment (antonwolfy) [inline dpnp/dpnp_container.py:208]: Shouldn't we use just:
```
return dpnp_array(array_obj.shape, buffer=array_obj, order="K")
```
will it guarantee the resulting usm_ndarray of the same order as passed by `buffer` here?

File: tests/test_arraycreation.py [hunk 9]
```diff
@@ -258,48 +258,44 @@ def test_tri_default_dtype():
 
 
 @pytest.mark.parametrize("k",
-                         [-6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6],
-                         ids=['-6', '-5', '-4', '-3', '-2', '-1', '0', '1', '2', '3', '4', '5', '6'])
+                         [-3, -2, -1, 0, 1, 2, 3, 4, 5],
```
Comment (antonwolfy) [inline tests/test_arraycreation.py:261]: Can we add here (and in test below) 0-dimension array of dpnp_array and usm_ndarray and numpy scalar?

File: tests/skipped_tests.tbl (hunk 2 above)
Comment (antonwolfy) [reply tests/skipped_tests.tbl:433]: can you add a fixture:
```
@pytest.mark.usefixtures("allow_fall_back_on_numpy")
```
before
```
class TestTriLowerAndUpper(unittest.TestCase):
```
to allow test to fallback on numpy with shape=(2,) rather then to silently skip it.

Any nested "allow_fall_back_on_numpy" fixtures above class methods can be removed then.

**Review by oleksandr-pavlyk — APPROVED** [review]
Looks good to me.

**Pull request merged**